doc/source/user/theory.broadcasting.rst

```diff
@@ -0,0 +1,229 @@
+:orphan:
+
+.. _array-broadcasting-in-numpy:
+
+===========================
+Array Broadcasting in Numpy
+===========================
+
+.. note::
+   Originally part of the scipy.org wiki, available `here
+   <https://scipy.github.io/old-wiki/pages/EricsBroadcastingDoc>`_ or from the
+   `github repo
+   <https://github.com/scipy/old-wiki/blob/gh-pages/pages/EricsBroadcastingDoc.html>`_
+
+[Let's] ... explore a more advanced concept in numpy called broadcasting. The
+term broadcasting describes how numpy treats arrays with different shapes
+during arithmetic operations. Subject to certain constraints, the smaller array
+is "broadcast" across the larger array so that they have compatible shapes.
+Broadcasting provides a means of vectorizing array operations so that looping
+occurs in C instead of Python. It does this without making needless copies of
+data and usually leads to efficient algorithm implementations. There are also
+cases where broadcasting is a bad idea because it leads to inefficient use of
+memory that slows computation. This article provides a gentle introduction to
+broadcasting with numerous examples ranging from simple to involved. It also
+provides hints on when and when not to use broadcasting.
+
+numpy operations are usually done element-by-element which requires two arrays
+to have exactly the same shape:
+
+.. code-block:: python
+    :caption: Example 1
+    :name: example-1
```

Comment:
all of these `code-block` and captions can simply be dropped.

Reply:
they provide an anchor for referencing them in the document, and provide nice formatting for copy-paste

Reply:
> they provide an anchor 

good point. okay, let's keep it as is.

> provide nice formatting for copy-paste

should be the same formatting as for a normal indented docstring example